azerothcore/azerothcore-wotlk
# feat(Scripting/Hooks): Implement OnQuestComputeXP() hook

## Changes Proposed:
-  Implement a hook called when quest XP is computed.
-  Allows for changing XP values for specific quests, quests in a level range or of a certain type, etc. I feel many may find it useful! There previously was no way to change XP for individual quest programmatically.

## Tests Performed:

- Compiles, simple test module correctly can call hook and change XP value.
- Completed several quests, ensured everything still works without breaking.


## How to Test the Changes:


1. Try calling the hook, make sure everything is working and values can be change.
2. Test completing various quest, make sure nothing is break.


## How to Test AzerothCore PRs
 
When a PR is ready to be tested, it will be marked as **[WAITING TO BE TESTED]**.

You can help by testing PRs and writing your feedback here on the PR's page on GitHub. Follow the instructions here:

http://www.azerothcore.org/wiki/How-to-test-a-PR

**REMEMBER**: when testing a PR that changes something **generic** (i.e. a part of code that handles more than one specific thing), the tester should not only check that the PR does its job (e.g. fixing spell XXX) but **especially** check that the PR does not cause any regression (i.e. introducing new bugs).

**For example**: if a PR fixes spell X by changing a part of code that handles spells X, Y, and Z, we should not only test X, but **we should test Y and Z as well**.
